src/useData.ts

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {
+    useState,
+    useEffect,
+} from 'react';
+import dataJson from './data/data.json';
+
+export interface IDataItem {
+    date: string;
+    open: number;
+    high: number;
+    low: number;
+    close: number;
+    volume: number;
+}
+
+export enum QueryType {
+    'RANGE',
+    'YTD',
+    '1Y',
+    '2Y',
+    // '3Y',
+    // '6M',
+    // '3M',
+    // '1M',
+    // '5D',
+    // '1D'
+}
+
+export interface IDataQuery {
+    queryString?: string;
+    queryType: QueryType;
+}
+
+function toTimestamp(dateString: string): number {
+    const [ d, m, y ] = dateString.split('/');
+    return new Date(`${m}/${d}/${y}`).getTime();
+}
+
+export const useData = (query: IDataQuery) => {
```

Comment:
So this is a fire & forget implementation of custom "fetch" hook. You pass the query, and you expect data to eventually be set. As query (passed to the component that uses the hook) channges it will rerender the component, and reuse the hook. The component is not interacting with the API at all.